darts/models/torch_forecasting_model.py

```diff
@@ -593,23 +597,28 @@ def predict_from_dataset(self,
                 batch_prediction = batch_prediction[:, :n, :]
                 batch_prediction = batch_prediction.cpu().detach().numpy()
                 
-                ts_forecasts = Parallel(n_jobs=n_jobs)(delayed(self._build_forecast_series)(prediction, input_series[0])
-                                                       for prediction, input_series in zip(batch_prediction,
-                                                                                           input_series_dataset))
+                ts_forecasts = Parallel(n_jobs=n_jobs)(
+                    delayed(self._build_forecast_series)(
+                        batch_prediction[batch_idx],
+                        input_series_dataset[dataset_idx][0]
+                    )
+                    for batch_idx, dataset_idx in enumerate(indices)
+                )
                 
                 predictions.extend(ts_forecasts)
 
         return predictions
 
-    def _prepare_predict_tensors(self, n: int, input_series_dataset: TimeSeriesInferenceDataset):
+    def _prepare_predict_timeseries(self, n: int, series, covariates):
         """
-        Creates a torch tensor `in_past` from `TimeSeriesInferenceDataset` instance for initial input to model
-        which includes the target series and covariate series if the model was trained with covariates.
-        Only past covariates are included, even if more are provided.
-
-        If the model is required to produce the forecast over multiple iterations, i.e. if
-        `n > self.output_chunk_length`, and if it was trained with covariates, then `cov_future` will
-        contain the future covariates up to `n` time steps into the future.
+        Creates 3 lists with `TimeSeries` instances required to produce model predictions. 
```

Comment:
Good comment, thanks!

Reply:
Adapted it and moved it to `SimpleInferenceDataset.__init__`